sale_loyalty_auto_refresh/tests/test_sale_loyalty_auto_refresh.py

```diff
@@ -0,0 +1,142 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import new_test_user, tagged
+
+from odoo.addons.sale_loyalty.tests.common import TestSaleCouponCommon
+
+
+@tagged("post_install", "-at_install")
+class TestSaleLoyaltyAutoRefresh(TestSaleCouponCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env["loyalty.program"].search([]).write({"active": False})
+        cls.env["loyalty.program"].create(
+            {
+                "name": "10% Discount",
+                "program_type": "coupons",
```

Comment:
```suggestion
                "program_type": "promotion",
```

Reply:
`coupons` can't be `auto` in v16: https://github.com/odoo/odoo/blob/16.0/addons/loyalty/models/loyalty_program.py#L195